File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/api/dto/BookUsing.java
package ru.spring5.elena.librarymanager.api.dto;

import lombok.Builder;
import lombok.Getter;

import java.util.Date;

@Builder
@Getter
public class BookUsing {
    private Long bookUsingId;
    private Date startDate; //Книгу взяли
    private Date endDate; //Книгу вернули
    private Long readerId; //кто взял книгу
    private Long bookId; //какую книгу взял
}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/dto/BookUsing.kt
package ru.elenacherev.librarymanager.api.dto

import java.time.Instant
import java.util.*

data class BookUsing(
    var bookUsingId: UUID? = null,
    val startDate: Instant,
    val endDate: Instant? = null,
    val readerId: UUID,
    val bookId: UUID
)